js/dropdownAdapter.js

```diff
@@ -0,0 +1,98 @@
+define([
+    'libraries/select2'
+], function() {
+	/*
+     * issue/1543: fix from https://github.com/select2/select2/issues/4063
+     */
+    var dropdownAdapter;
+    jQuery.fn.select2.amd.require([
+        "select2/utils",
+        "select2/dropdown",
+        "select2/dropdown/attachContainer",
+        "select2/dropdown/closeOnSelect"
+    ], function(Utils, DropdownAdapter, AttachContainer, CloseOnSelect) {
+
+        /*
+         * issues/1889: fix from https://github.com/adaptlearning/adapt_framework/issues/1889
+         * code added to make dropdown sit above body bottom where appropriate
+         */
+        AttachContainer.prototype.bind = function(decorated, container, $container) {
+
+            decorated.call(this, $container, $container);
+
+            container.on('opening', function () {
+                // hide so that popup doesn't jump when repositioned
+                this.$dropdown.css("visibility", "hidden");
+            }.bind(this));
+
+            container.on('open', function () {
+                // add dropdown at this point so that the browser focus works correctly
+                var $dropdownContainer = $container.find('.dropdown-wrapper');
+                $dropdownContainer.append(this.$dropdown);
+
+                // defer to allow dom to settle before repositioning
+                setTimeout(function() {
+                    this.position(this.$dropdown, $container);
+                    this.$dropdown.css("visibility", "");
+                }, 0);
```

Comment:
Do we need a context binding here?

Reply:
@danielstorey thanks for reviewing and partiularly for spotting this. Dropped the reference to `self` and must have intended to use `Function.bind` but forgot :-/